apps/home/routes.py

```diff
@@ -26,16 +26,71 @@ def get_info_before_request():
 @blueprint.route('/index')
 @login_required
 @check_user_category(["admin", "teacher", "student"])
+@cache.cached(timeout=300)
 def index():
+    lab_instances = LabInstances.query.filter_by(is_deleted=False).count()
+    users = Users.query.filter_by(is_deleted=False).count()
+    labs = Labs.query.count()
+
+    try:
+        k8s.update_nodes()
+        total_cpu_capacity = 0
+        total_cpu_usage = 0
+        total_memory_capacity = 0
+        total_memory_usage = 0
+        total_storage_capacity = 0
+        total_storage_usage = 0
+        total_pods = 0
+        total_nodes = len(k8s.ready_nodes)
+
+        for node_name in k8s.ready_nodes:
+            node = k8s.nodes.get(node_name)
+            if node:
+                # CPU
+                cpu_capacity = int(node.status.capacity.get("cpu", 0))
+                total_cpu_capacity += cpu_capacity
+                cpu_usage = int(node.status.allocatable.get("cpu", 0))
+                total_cpu_usage += cpu_usage
+
+                # Memory
+                memory_capacity = int(node.status.capacity.get("memory", "0").rstrip("Ki")) // (1024*1024)
+                total_memory_capacity += memory_capacity
+                memory_usage = int(node.status.allocatable.get("memory", "0").rstrip("Ki")) // (1024*1024)
+                total_memory_usage += memory_usage
+
+                # Storage
+                storage_capacity = int(node.status.capacity.get("ephemeral-storage", "0").rstrip("Ki")) // (1024*1024)
+                total_storage_capacity += storage_capacity
+                storage_usage = int(node.status.allocatable.get("ephemeral-storage", "0").rstrip("Ki")) // (1024*1024)
+                total_storage_usage += storage_usage
+
+                # Pods
+                pods_capacity = int(node.status.capacity.get("pods", 0))
+                total_pods += pods_capacity
```

Comment:
we should move this over the kubernetes controller class (`apps/controllers/kubernetes.py`) and abstract this specific computation there, only returning a dict with the needed information. Can you please refactor this part?